restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/impl/OrExpressionDocumentsResolver.java

```diff
@@ -186,6 +186,7 @@ public OrExpressionDocumentsResolver(Or<FilterExpression> expression, ExecutionC
 
   private ExecutionContext createContext(
       ExecutionContext context, Or<FilterExpression> expression) {
-    return context.nested("MERGING OR: expression '" + expression + "'");
+    // Note: use toLexicographicString to ensure a stable order of sub-expressions.
+    return context.nested("MERGING OR: expression '" + expression.toLexicographicString() + "'");
```

Comment:
This is needed to unbreak the unit tests now that the `.toString()` representation became unstable due to expression transformations in `ExpressionUtils`.